sources/packages/backend/apps/queue-consumers/src/services/student-file/student-file.service.ts

```diff
@@ -27,37 +28,44 @@ export class StudentFileService extends RecordDataModelService<StudentFile> {
   /**
    * Scans the file with the provided unique filename
    * for any viruses.
-   * @param uniqueFileName unique filename of the file to perform the virus scan.
+   * @param jobData virus scan job data.
+   * @param attemptsMade number of attempts made to run this job.
    * @param processSummary process summary logs.
    * @returns boolean true if the file is virus infected, false otherwise.
    */
   async scanFile(
-    uniqueFileName: string,
+    jobData: VirusScanQueueInDTO,
+    attemptsMade: number,
     processSummary: ProcessSummary,
   ): Promise<boolean> {
-    const studentFile = await this.getStudentFile(uniqueFileName);
+    const studentFile = await this.getStudentFile(jobData.uniqueFileName);
     if (!studentFile) {
-      throw new NotFoundException(`Student file ${uniqueFileName} not found.`);
+      throw new CustomNamedError(
+        `Student file ${jobData.uniqueFileName} not found.`,
+        FILE_NOT_FOUND,
+      );
     }
+
     const stream = new Readable();
     stream.push(studentFile.fileContent);
     stream.push(null);
-    const isInfected = await this.clamAVService.scanFile(
-      stream,
-      processSummary,
-    );
-    if (isInfected == null) {
-      // If the file could not be scanned for any reason,
-      // update the file scan status to Pending to keep a
-      // track of the files that haven't been scanned yet.
-      await this.updateFileScanStatus(studentFile.uniqueFileName, isInfected);
+    const virusScanCode = await this.clamAVService.scanFile(stream);
+    if (virusScanCode.errorCode || virusScanCode.isInfected === null) {
```

Comment:
As per you clarify in our last chat, `null` is also expected and it will be considered as error to be retried right now.
It is also present in the docs as a possible value: [scanStream](https://github.com/kylefarris/clamscan?tab=readme-ov-file#returns-4)